src/data.rs

```diff
@@ -1427,6 +1433,16 @@ unsafe impl<'a> Binding<'a> for DataNodeRef<'a> {
     ) -> DataNodeRef<'a> {
         DataNodeRef { tree, raw }
     }
+
+    unsafe fn into_raw(self) -> *mut Self::CType {
```

Comment:
This should probably be a safe function, as there's no inherent risk in the conversion (similar to `Box::from_raw` being unsafe and `Box::into_raw` being safe).